File: src/main/java/editor/gui/ccp/handler/ImportHandler.java
package editor.gui.ccp.handler;

import java.awt.datatransfer.DataFlavor;

import javax.swing.TransferHandler;

/**
 * Interface describing a handler for importing a single flavor of data.
 * 
 * @author Alec Roelke
 */
public interface ImportHandler
{
    /**
     * @return The data flavor supported by this ImportHandler.
     */
    DataFlavor supportedFlavor();

    /**
     * @see TransferHandler#canImport
     */
    boolean canImport(TransferHandler.TransferSupport supp);
    
    /**
     * @see TransferHandler#importData
     */
    boolean importData(TransferHandler.TransferSupport supp);
}

File: src/main/scala/editor/gui/ccp/handler/ImportHandler.scala
package editor.gui.ccp.handler

import java.awt.datatransfer.DataFlavor
import javax.swing.TransferHandler

/**
 * Transfer handler that supports importing data of a single flavor.
 * @author Alec Roelke
 */
trait ImportHandler extends TransferHandler {
  /** @return the data flavor supported by the handler */
  def supportedFlavor: DataFlavor
}